cmd/process-agent/command/command.go

```diff
@@ -106,18 +110,89 @@ func MakeCommand(subcommandFactories []SubcommandFactory, winParams bool, rootCm
 }
 
 // BootstrapConfig is a helper for process-agent config initialization (until we further refactor to use components)
-func BootstrapConfig(globalParams *GlobalParams) error {
-	if err := pconfig.LoadConfigIfExists(globalParams.ConfFilePath); err != nil {
+func BootstrapConfig(path string, oneshotCommand bool) error {
+	setHostMountEnv()
+
+	if err := loadConfigIfExists(path); err != nil {
+		return err
+	}
+
+	// Resolve any secrets
+	if err := config.ResolveSecrets(config.Datadog, filepath.Base(path)); err != nil {
 		return err
 	}
 
+	var (
+		logFile      string
+		syslogURI    string
+		syslogRFC    = false
+		logToConsole = true
+		logAsJSON    = false
+	)
+
+	if !oneshotCommand {
+		syslogURI = config.GetSyslogURI()
+		syslogRFC = config.Datadog.GetBool("syslog_rfc")
+		logToConsole = config.Datadog.GetBool("log_to_console")
+		logAsJSON = config.Datadog.GetBool("log_format_json")
+
+		if !config.Datadog.GetBool("disable_file_logging") {
+			logFile = config.Datadog.GetString("process_config.log_file")
+		}
+	}
+
 	return config.SetupLogger(
 		LoggerName,
 		config.Datadog.GetString("log_level"),
-		"",
-		"",
-		false,
-		true,
-		false,
+		logFile,
+		syslogURI,
+		syslogRFC,
+		logToConsole,
+		logAsJSON,
 	)
 }
+
+// setHostMountEnv sets HOST_PROC and HOST_SYS mounts if applicable in containerized environments
+func setHostMountEnv() {
+	// Set default values for proc/sys paths if unset.
+	// Don't set this is /host is not mounted to use context within container.
+	// Generally only applicable for container-only cases like Fargate.
+	if !config.IsContainerized() {
+		return
+	}
+
+	_, err := os.Stat("/host")
+	if !os.IsNotExist(err) {
```

Comment:
Not sure how much of an impact this has, but it seems like there is a small change in behavior - previously we were using `util.PathExists`, which return true only if there was no error returned from `os.Stat`, whereas here we continue if `!os.IsNotExist`. In particular it seems like if the error was something like a permission error we'd see different behavior compared to the previous implementation which would not set these environment variables, whereas this one would

Reply:
Yep, thank you, nice catch! To think of it, the helper exists in the util which will still be around for some time, no reason not to use it.

Reply:
Sounds good, looks like there is another occurrence of this in `loadConfigIfExists` below as well